packages/react-compose/src/mergeComposeOptions.ts

```diff
@@ -42,9 +43,10 @@ export function mergeComposeOptions(
       });
       return mergedSlotProps;
     }, {});
-  return {
+
+  const preparedOptions: Partial<ComposePreparedOptions> = {
     className: inputOptions.className || parentOptions.className,
-    classes: mergeClasses(parentOptions.classes, inputOptions.classes),
+    classes: [...parentOptions.classes, inputOptions.classes],
```

Comment:
i don't like how we are re-resolving all classes from all compose layers. user should be able to replace/avoid styles calculation from parent components if the composed component doesn't need